scripts/compile-dev.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash -e
+
+set -ex
+repo=intel/graph-compiler
+
+cd $(dirname "$0")/..
+project_dir=$PWD
+llvm_hash=$(cat cmake/llvm-version.txt)
+
+mkdir -p externals
+pushd externals
+
+if ! [ -d "llvm-project" ]; then
+    git clone https://github.com/llvm/llvm-project.git
+fi
+
+pushd llvm-project
+git checkout ${llvm_hash}
+
+cmake -G Ninja llvm -B build \
+    -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_ASSERTIONS=true \
+    -DLLVM_ENABLE_PROJECTS="mlir" -DLLVM_TARGETS_TO_BUILD="X86" \
+    -DLLVM_INSTALL_UTILS=true -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+    -DLLVM_INSTALL_GTEST=ON
+cmake --build build 
+echo $project_dir
+mlir_dir=$PWD/build/lib/cmake/mlir
+popd
+popd
```

Comment:
I consider using `pushd` / `popd` bad practice, they are not local. Use `cd` instead. Another option is to use round brackets to do something in a separate shell, then you return to the default dir